indico/web/util.py

```diff
@@ -179,3 +181,34 @@ def get_request_info(hide_passwords=True):
 def url_for_index(_external=False, _anchor=None):
     from indico.web.flask.util import url_for
     return url_for('categories.display', _external=_external, _anchor=_anchor)
+
+
+def signed_url_for(user, blueprint, url_params=None, *args, **kwargs):
+    from indico.web.flask.util import url_for
+    _external = kwargs.pop('_external', False)
+    base_url = url_for(blueprint, *args, **(url_params or {}))
+    qs = url_encode(sorted(kwargs.items()))
+    url = '{}?{}'.format(base_url, qs) if qs else base_url
+    signer = Signer(user.signing_secret, salt='url-signing')
+    print 'signing', url
```

Comment:
:eyes: :printer:

Reply:
Debugging code I added because of the unit tests. We should really add a pre-commit hook to look for print and console ☹️